nucliadb_node/src/merge/scheduler.rs

```diff
@@ -0,0 +1,368 @@
+// Copyright (C) 2021 Bosutech XXI S.L.
+//
+// nucliadb is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at info@nuclia.com.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+use std::fs;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Condvar;
+use std::time::Duration;
+use std::{
+    path::PathBuf,
+    sync::{Arc, Mutex},
+};
+
+use anyhow::anyhow;
+use nucliadb_core::vectors::MergeMetrics;
+use tokio::sync::oneshot;
+use tokio::sync::oneshot::{Receiver, Sender};
+
+use crate::shards::ShardId;
+use crate::{settings::Settings, shards::providers::shard_cache::ShardWriterCache};
+use nucliadb_core::tracing::warn;
+use nucliadb_core::NodeResult;
+
+use super::work::WorkQueue;
+use crate::merge::{MergePriority, MergeRequest, MergeWaiter};
+
+// Time between scheduler being idle and scheduling all shards for merge.
+//
+// This value also affects shutdown reaction time
+#[cfg(not(test))]
+const SCHEDULE_ALL_SHARDS_DELAY: Duration = Duration::from_secs(10);
+#[cfg(test)]
+const SCHEDULE_ALL_SHARDS_DELAY: Duration = Duration::from_millis(50);
```

Comment:
Maybe this should be an attribute of the scheduler, to avoid the conditional compilation.
Not important though, just if you think is worth it.

Reply:
I can add as a setting and grab it from there too